src/main/webapp/app/core/util/alert.service.ts

```diff
@@ -121,23 +127,38 @@ export class AlertService {
      *                   If `translateKey` is available then it's translation else `message` is used for showing.
      * @returns  Added alert
      */
-    addAlert(alert: AlertBase): Alert {
-        const alertInternal = alert as AlertInternal;
-        alertInternal.id = this.alertId++;
+    addAlert(alert: AlertCreationProperties): Alert {
+        // Defensive copy to prevent overwrites
+        const alertInternal = { ...alert } as AlertInternal;
```

Comment:
Not sure at the moment how it works in Typescript, could you pass a normal object but expect a Readonly of it?

=> `addAlert(alert: Readonly<AlertCreationProperties>): Alert {}`
and then `const newAlert: AlertCreationProperties = {}; service.addAlert(newAlert);`

If that works it would be better than just copying but I can't test it myself right now.

Reply:
People would still be able to change fields on the object, e.g.
```js
const newAlert: any = {
    ...
}
service.addAlert(newAlert as AlertCreationProperties);
newAlert.message = ...
```
If we copy the object and someone runs the code above, nothing would happen as they are modifying the original object (and not our copy). The returned object from the ``addAlert`` call cannot be legally modified (at least TS will complain about it).

The other 'alternative' to creating a copy would be freezing the object; but that would prevent the service itself from updating the alert, i.e. the 'isOpen' property.

Any other ideas? :D

Reply:
I see, it's not as easy as I imagined. It's not important, let's keep it like it is.